modules/light-clients/08-wasm/types/proposal_handle_test.go

```diff
@@ -78,202 +84,99 @@ func (suite *TypesTestSuite) TestCheckSubstituteAndUpdateStateGrandpa() {
-// 			substituteWasmClientState = suite.chainA.GetClientState(substitutePath.EndpointA.ClientID).(*types.ClientState)
-// 			err = suite.chainA.Codec.UnmarshalInterface(substituteWasmClientState.Data, &substituteWasmClientStateData)
-// 			suite.Require().NoError(err)
-// 			substituteTmClientState = substituteWasmClientStateData.(*ibctm.ClientState)
-
-// 			// test that subject gets updated chain-id
-// 			newChainID := "new-chain-id"
-// 			substituteTmClientState.ChainId = newChainID
+func (suite *TypesTestSuite) TestCheckSubstituteAndUpdateState() {
+	var substituteClientState exported.ClientState
+	contractErr := errors.New("contract error")
 
-// 			substituteTmClientStateBz, err = clienttypes.MarshalClientState(suite.chainA.App.AppCodec(), substituteTmClientState)
-// 			suite.Require().NoError(err)
-// 			substituteWasmClientState.Data = substituteTmClientStateBz
-// 			substitutePath.EndpointA.SetClientState(substituteWasmClientState)
-
-// 			subjectClientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), subjectPath.EndpointA.ClientID)
-// 			substituteClientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), substitutePath.EndpointA.ClientID)
-
-// 			expectedConsState := substitutePath.EndpointA.GetConsensusState(substituteWasmClientState.GetLatestHeight())
-// 			expectedProcessedTime, found := ibctm.GetProcessedTime(substituteClientStore, substituteWasmClientState.GetLatestHeight())
-// 			suite.Require().True(found)
-// 			expectedProcessedHeight, found := GetProcessedHeight(substituteClientStore, substituteWasmClientState.GetLatestHeight())
-// 			suite.Require().True(found)
-// 			expectedIterationKey := ibctm.GetIterationKey(substituteClientStore, substituteWasmClientState.GetLatestHeight())
+	testCases := []struct {
+		name     string
+		malleate func()
+		expErr   error
+	}{
+		{
+			"success",
+			func() {
+				suite.mockVM.RegisterSudoCallback(
+					types.CheckSubstituteAndUpdateStateMsg{},
+					func(_ cosmwasm.Checksum, _ wasmvmtypes.Env, sudoMsg []byte, store cosmwasm.KVStore, _ cosmwasm.GoAPI, _ cosmwasm.Querier, _ cosmwasm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+						var payload types.SudoMsg
+						err := json.Unmarshal(sudoMsg, &payload)
+						suite.Require().NoError(err)
+
+						suite.Require().NotNil(payload.CheckSubstituteAndUpdateState)
+						suite.Require().Nil(payload.UpdateState)
+						suite.Require().Nil(payload.UpdateStateOnMisbehaviour)
+						suite.Require().Nil(payload.VerifyMembership)
+						suite.Require().Nil(payload.VerifyNonMembership)
+						suite.Require().Nil(payload.VerifyUpgradeAndUpdateState)
+
+						bz, err := json.Marshal(types.EmptyResult{})
+						suite.Require().NoError(err)
+
+						return &wasmvmtypes.Response{Data: bz}, types.DefaultGasUsed, nil
+					},
+				)
+			},
+			nil,
+		},
+		{
+			"failure: invalid substitute client state",
```

Comment:
a test case will need to be added here for the code hash equality check

Reply:
noice, thanks for pointing these out :1st_place_medal: